### PR TITLE
fixed bytecode compaction for ExecuteSC

### PR DESCRIPTION
**Pull request by BatiGencho**
This MR fixes the bytecode compaction for the op datastore added to v15. It also revises the sc test and introduces other small fixes.

### CODE REVIEW

File: src/interfaces/OperationTypes.ts [hunk 1]
```diff
@@ -23,6 +23,7 @@ export interface IExecSmartContractOpType {
         max_gas: number, // maximum amount of gas that the execution of the contract is allowed to cost.
         coins: string, // represent an Amount in coins that are spent by consensus and are available in the execution context of the contract.
```
Comment (AurelienFT) [inline src/interfaces/OperationTypes.ts:24]: Coins shouldn't be here anymore

File: src/web3/BaseClient.ts [hunk 2]
```diff
@@ -167,17 +167,17 @@ export class BaseClient {
...
 				const maxGasEncoded = Buffer.from(varintEncode((data as IContractData).maxGas));
 
-				// coins to send
-				const coinsEncoded = Buffer.from(varintEncode((data as IContractData).coins));
-
 				// gas price
 				const gasPriceEncoded = Buffer.from(varintEncode((data as IContractData).gasPrice));
 
+				// sc local datastore
+				const scLocalDatastore = Buffer.from(new Uint8Array());
```
Comment (AurelienFT) [inline src/web3/BaseClient.ts:174]: This is default value when not taking real value ?

File: src/interfaces/OperationTypes.ts [hunk 3]
```diff
@@ -20,9 +20,9 @@ export interface IRollSellOpType {
 export interface IExecSmartContractOpType {
     ExecuteSC: {
         data: [number], // vec of bytes to execute
-        max_gas: number, // maximum amount of gas that the execution of the contract is allowed to cost.
-        coins: string, // represent an Amount in coins that are spent by consensus and are available in the execution context of the contract.
+        max_gas: number, // maximum amount of gas that the execution of the contract is allowed to cost
         gas_price: string, // represent an Amount in coins, price per unit of gas that the caller is willing to pay for the execution.
+        datastore: any,
```
Comment (AurelienFT) [inline src/interfaces/OperationTypes.ts:25]: Can we change the type here to a concrete one ?

Comment (BatiGencho) [reply src/interfaces/OperationTypes.ts]: done

**Review by AurelienFT — APPROVED** [review]

**Comment by qdrn** [comment]
Thanks @AurelienFT !

**Pull request merged**